desktop/src/renderer/src/pages/FolderList.tsx

```diff
@@ -0,0 +1,196 @@
+import {
+	ContinueButton,
+	ContinueModal,
+	FolderDisplayGrid,
+	type FolderRow,
+	SelectFolderButton,
+} from '@renderer/components/file-list';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { TokenContext } from '../App';
+
+type Props = {
+	authenticated: boolean;
+};
+
+export const FolderListPage = ({ authenticated }: Props) => {
+	const [continueButtonIsEnabled, setContinueButtonIsEnabled] =
+		useState<boolean>(false);
+	const [continueModalIsOpen, setContinueModalIsOpen] =
+		useState<boolean>(false);
+
+	const accessToken = useContext(TokenContext);
+
+	const theme = useTheme();
+	const apiRef = useGridApiRef();
+	const folderList = useFolderList({ accessToken });
+
+	const handleProgress = useCallback(
+		(event: CustomEvent<{ source: string; progressPercentage: number }>) => {
+			const { source, progressPercentage } = event.detail;
+			folderList.setFolders((prevFolderList) =>
+				prevFolderList.map((folder) =>
+					folder.folder === source
+						? { ...folder, progress: progressPercentage }
+						: folder,
+				),
+			);
+		},
+		[folderList.setFolders],
+	);
+
+	const handleCompletion = useCallback(
+		(
+			event: CustomEvent<{
+				source: string;
+				success: boolean;
+				metadata?: Record<string, unknown>;
+				error?: unknown;
+			}>,
+		) => {
+			const { source, success, metadata: newMetadata } = event.detail;
+
+			if (success && newMetadata) {
+				folderList.setMetaData((prev) => ({
+					...prev,
+					[source]: newMetadata[source],
+				}));
+				console.info(`Successfully processed folder: ${source}`);
+			} else {
+				console.error(`Failed to process folder: ${source}`);
+			}
+		},
+		[folderList.setMetaData],
+	);
+
+	const handleRowUpdate = useCallback(
+		(newFolder: FolderRow) => {
+			folderList.setFolders((prevFolderList) =>
+				prevFolderList.map((folder) =>
+					folder.id === newFolder.id ? newFolder : folder,
+				),
+			);
+			return newFolder;
+		},
+		[folderList.setFolders],
+	);
+
+	const handleOpenContinueModel = useCallback(() => {
+		let isOpen = true;
+
+		if (!continueButtonIsEnabled) isOpen = false;
+
+		setContinueModalIsOpen(isOpen);
+	}, [continueButtonIsEnabled]);
+
+	const handleClose = useCallback(() => {
+		setContinueModalIsOpen(false);
+	}, []);
+
+	const handleFormSubmit = useCallback(
+		(formData) => {
+			folderList.submit(formData);
+		},
+		[folderList.submit],
+	);
+
+	useEffect(() => {
+		window.addEventListener(
+			'folder-metadata-progress',
+			handleProgress as EventListener,
+		);
+
+		window.addEventListener(
+			'folder-metadata-completion',
+			handleCompletion as EventListener,
+		);
+
+		return () => {
+			window.removeEventListener(
+				'folder-metadata-progress',
+				handleProgress as EventListener,
+			);
+			window.removeEventListener(
+				'folder-metadata-completion',
+				handleCompletion as EventListener,
+			);
+		};
+	}, [handleCompletion, handleProgress]);
+
+	useEffect(() => {
+		let isEnabled = true;
```

Comment:
Theres no need to do a bunch of if statements here. You can just make isEnabled a boolean and set it to `folderList.folders.length > 0 && authenticated && folderList.folders.every((folder) => folder.progress === 100)`